File: week_2/8_led_control_add.cpp
int buttonPin = 3;
int ledPin = 11;

void setup() {
    pinMode(buttonPin, INPUT);
    pinMode(ledPin, OUTPUT);
}

void loop() {
    if (digitalRead(buttonPin) == HIGH) {
        for (int i = 0; i <= 255; i += 5) {
            analogWrite(ledPin, i);
            delay(10);
        } else {
            analogWrite(ledPin, 0);
        }
    }

    // 밝기 증가 감소 간격 변경
    // for (int brightness = 0; brightness <= 255; brightness++) {
    //     analogWrite(ledPin, brightness);
    //     delay(10);
    // }
    // for (int brightness = 255; brightness >= 0; brightness--) {
    //     analogWrite(ledPin, brightness);
    //     delay(10);
    // }
}

File: week_2/8_led_control_add.py
from Arduino import Arduino
import time

ledPin = 11

board = Arduino("9600", port = "COM3")
board.pinMode(ledPin, "OUTPUT")

while True:
    brightness = 0
    while brightness < 255:
        board.analogWrite(ledPin, brightness)
        brightness += 1
        time.sleep(0.01)
    while brightness > 0:
        board.analogWrite(ledPin, brightness)
        brightness -= 1
        time.sleep(0.01)
